gossip-base/src/main/java/org/apache/gossip/manager/GossipMemberStateRefresher.java

```diff
@@ -112,10 +115,16 @@ public boolean processOptimisticShutdown(Entry<LocalMember, GossipState> l) {
     if (s.getShutdownAtNanos() > l.getKey().getHeartbeat()) {
       members.put(l.getKey(), GossipState.DOWN);
       if (l.getValue() == GossipState.UP) {
-        listener.gossipEvent(l.getKey(), GossipState.DOWN);
+        for (GossipListener listener: listeners)
```

Comment:
Do these in a callable/executor. I do not want to risk arbitrary code blocking gossip internals.